utils/http_client.py

```diff
@@ -31,8 +31,9 @@ async def fetch_data(self, *args, **kwargs) -> str | None:
         try:
             async with self.session.get(*args, **kwargs) as response:
                 # 404 response is expected for some requests.
-                if response.status != 404:
+                if response.status == 200:
                     return await response.text()
+            return 
```

Comment:
no need to return anything here

Reply:
In order to keep the code consistent, could you match the code used in post_data please. Aka, use `match case` statement, and similarly log f"Fetch request error: (code: {response.status})" if the status code is not 200

Reply:
In order to keep the code consistent, could you match the code used in post_data please. Aka, use `match case` statement, and similarly log a f"Fetch request error: (code: {response.status})" message if the status code is not 200